ipa-core/src/protocol/ipa_prf/dp_in_mpc/dp_in_mpc.rs

```diff
@@ -0,0 +1,100 @@
+// DP in MPC
+
+use futures_util::{stream, StreamExt};
+
+// use ipa_macros::Step;
+use crate::ff::{CustomArray, U128Conversions};
+use crate::{
+    error::Error,
+    ff::boolean::Boolean,
+    protocol::{
+        context::{Context, UpgradedSemiHonestContext},
+        ipa_prf::aggregation::aggregate_values,
+        prss::{FromPrss, SharedRandomness},
+        BooleanProtocols, RecordId,
+    },
+    secret_sharing::{
+        replicated::semi_honest::{AdditiveShare as Replicated, AdditiveShare},
+        BitDecomposed, FieldSimd, SharedValue, Vectorizable,
+    },
+    sharding::NotSharded,
+};
+
+#[cfg(test)]
+pub async fn gen_binomial_noise<'ctx, const B: usize, OV>(
+    ctx: UpgradedSemiHonestContext<'ctx, NotSharded, Boolean>,
+    _never_used_input: BitDecomposed<Replicated<Boolean, B>>,
```

Comment:
which part were you referring to as worrying? the cost of aggregation for this could get expensive if the privacy parameters are very small (like epsilon = 0.01 scale).  You can play with this spreadsheet I just added you to as well to see how it looks. https://docs.google.com/spreadsheets/d/1sMgqkMw3-yNBp6f8ctyv4Hdfx9Ei7muj0ZhP9i1DHrw/edit#gi 

For delta = 10^{-8}
* epsilon = 1, need to aggregate 221 Bernoullis
* epsilon = 0.1, need to aggregate 22k Bernoullis
* epsilon = 0.01, need to aggregate 2.2M Bernoullis